crates/cadenza-eval/src/value.rs

```diff
@@ -0,0 +1,200 @@
+//! Runtime value representation for the Cadenza evaluator.
+//!
+//! Values can be symbols, lists, functions, macros, or built-in operations.
+
+use crate::error::Result;
+use crate::interner::InternedId;
+use std::fmt;
+
+/// A runtime value in the Cadenza evaluator.
+#[derive(Clone)]
+pub enum Value {
```

Comment:
All of these need syntax nodes attached to them so we know how to attribute errors and build stack traces.

We also need to have a syntax tree be a runtime value so we can operate on it in use space.

I think we also want types to be eval-time values so they can be operated on and inspected.